### PR TITLE
Enable Dependabot for GitHub Actions

### LINKED ISSUE #17743: Enable Dependabot for GitHub Actions
### Summary of the new feature / enhancement

This repository already uses Dependabot but not for GitHub Actions used in workflows. I suggest enabling it to keep Actions updated.

I see some old (major) versions used in various workflows, with CodeQL v1 being probably the most important one because it will be deprecated in December.
https://github.blog/changelog/2022-04-27-code-scanning-deprecation-of-codeql-action-v1/

### Proposed technical implementation details (optional)

Adding it is pretty simple, just append [`dependabot.yml`](https://github.com/PowerShell/PowerShell/blob/bff54ba38dd3c2dac6d2fe5ebbdfa87d79df312e/.github/dependabot.yml):

```yml
  - package-ecosystem: "github-actions"
    directory: "/"
    schedule:
      interval: "daily"
```

### PR DESCRIPTION
**Pull request by Molkree**
# PR Summary

Enable Dependabot for GitHub Actions.

## PR Context

This repository already uses Dependabot but not for GitHub Actions used in workflows.

I see some old (major) versions used in various workflows, with CodeQL v1 being probably the most important one because it will be deprecated in December.
https://github.blog/changelog/2022-04-27-code-scanning-deprecation-of-codeql-action-v1/

Fixes #17743

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -51,3 +51,8 @@ updates:
       interval: "daily"
     labels:
       - "CL-BuildPackaging"
+
+  - package-ecosystem: "github-actions"
+    directory: "/"
```
Comment (daxian-dbw) [inline .github/dependabot.yml:56]: Shouldn't this be `/.github/workflows`? All GitHub actions are deployed in that folder.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -51,3 +51,8 @@ updates:
...
+    schedule:
+      interval: "daily"
```
Comment (daxian-dbw) [inline .github/dependabot.yml:58]:
```suggestion
    schedule:
      interval: "daily"
    labels:
      - "CL-BuildPackaging"
```

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (Molkree) [reply .github/dependabot.yml:56]: Docs say to set it to `/` for GitHub Actions ([1](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot#:~:text=Set%20the%20directory%20to%20%22/%22%20to%20check%20for%20workflow%20files%20in%20.github/workflows.), [2](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#directory)).

> Set the directory to "/" to check for workflow files in .github/workflows.

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Pull request merged**